sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -470,35 +495,25 @@ def refresh(self, **kwargs) -> None:
         need_refresh = False
         try:
             updated_sentinel_keys = dict(self._refresh_on)
+            feature_flag_sentinel_keys = dict(self._refresh_on_feature_flags)
             headers = _get_headers("Watch", uses_key_vault=self._uses_key_vault, **kwargs)
-            for (key, label), etag in updated_sentinel_keys.items():
-                try:
-                    updated_sentinel = self._client.get_configuration_setting(  # type:ignore
-                        key=key,
-                        label=label,
-                        etag=etag,
-                        match_condition=MatchConditions.IfModified,
-                        headers=headers,
-                        **kwargs
+            if self._refresh_on:
+                for (key, label), etag in updated_sentinel_keys.items():
```

Comment:
Could combine these for loops using one dict containing all sentinel keys (refresh_on and refresh_on_feature_flags).  Could still do the `if updated_sentinel is not None:` check if encased in an if with `~~~ in updated_sentinel_keys`

Reply:
Feature Flags and Configuration settings shouldn't cause the other to refresh. The whole section is wrong.